blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -202,27 +272,29 @@ namespace blueprint
         void dispatchViewEvent (ViewId viewId, const juce::String& eventType, T... args)
         {
             JUCE_ASSERT_MESSAGE_THREAD
-            jassert(engine);
-            
-            // engine may have been reset in the event of a bundle eval error. We don't want to
-            // crash if client code tries to immediately dispatch an event after bundle loading.
-            // ReactApplicationRoot should report a generic error to the user rather than crashing.
-            if (engine)
-                engine->invoke("__BlueprintNative__.dispatchViewEvent", viewId, eventType, std::forward<T>(args)...);
+            jassert(bundleValid);
+
+            // We wish to safe guard against client code which attempts to dispatch an event immediately
```

Comment:
Yea this seems like a consequence of not having a consistent error handling story. Glad we're thinking about this now

Reply:
Yep totally. Should be able to remove this with the exception handling approach.